app/controllers/staff/activities_controller.rb

```diff
@@ -39,7 +41,25 @@ def id
   end
 
   def organisation_id
-    params[:organisation_id]
+    organisation_id = params.fetch(:organisation_id, current_user.organisation)
+
+    return current_user.organisation.id unless known_organisation_id?(organisation_id)
+    return current_user.organisation.id if organisation_id.blank?
+    return current_user.organisation.id if requested_organisation_is_not_current_users?(organisation_id) && current_user_is_not_service_owner?
+    organisation_id
+  end
+
+  def requested_organisation_is_not_current_users?(requested_organisation_id)
+    requested_organisation_id != current_user.organisation.id
+  end
+
+  def current_user_is_not_service_owner?
+    return false if current_user.organisation.service_owner?
+    true
+  end
+
+  def known_organisation_id?(organisation_id)
+    Organisation.all.pluck(:id).include?(organisation_id)
```

Comment:
The `exists?` method might remove some logic here and perhaps prevent the need to construct a new hash of ids with `pluck`.

```
Organisation.exists?(organisation_id)
```

Arguably it's now succint enough not to need it it's own method?

Reply:
Lovely! Thank you.

Reply:
@tahb I am getting a brakeman sql injection warning with `exists?` any thoughts? Put an ignore in rethink? I would have thought Rails would make the actual SQL safe?

Reply:
Looking into this, it seems the accepted answer is to make sure the id is a string: 

https://github.com/presidentbeef/brakeman/issues/1045

Which seems to make brakeman happy.

Reply:
@tahb if you are happy with ☝️ that - I can merge this in – I want to make sure you have at least seen this as it was a potential security issue.

Reply:
Thanks for flagging.

I wondered even with `.to_s` you could craft a malicious string to escape out and inject more into the query. https://rails-sqli.org/rails4#exists does say that `.to_s` is a way to make it safe though which is good enough for me.

> To be be safe, convert user input to a string or integer if using it as the primary key in exists?.

The risk is that we accidentally remove the `.to_s` later without realising however Brakeman is seemingly ready and waiting to let us know 👍